src/coord/timestamp.rs

```diff
@@ -1396,14 +1426,19 @@ impl Timestamper {
         _id: SourceInstanceId,
         fc: &FileSourceConnector,
         timestamp_topic: String,
+        max_ts_batch: i64,
     ) -> Option<ByoFileConnector<avro::types::Value, failure::Error>> {
         let ctor = move |file| avro::Reader::new(file);
         let tail = if fc.tail {
             FileReadStyle::TailFollowFd
         } else {
             FileReadStyle::ReadOnce
         };
-        let (tx, rx) = std::sync::mpsc::sync_channel(self.max_increment_size as usize);
+        let (tx, rx) = if max_ts_batch > 0 {
+            std::sync::mpsc::sync_channel(max_ts_batch as usize)
+        } else {
+            std::sync::mpsc::sync_channel(10000 as usize)
```

Comment:
Does 0 mean "unbounded" or "default of 10000"? The code currently seems to do a bit of both?

Reply:
sync_channel doesn't seem to allow us to specify an unbounded option, unless I missed it? This is for files specifically, where we have a separate thread that has to communicate with the timestamping thread. My understanding is that if the channel is full, the producer will block sending the data in?

Reply:
Oh, I see; you probably want `std::sync::mpsc::channel`, but that'll make the types annoying.